src/diffusers/models/unet_2d_blocks.py

```diff
@@ -2293,11 +2306,32 @@ def __init__(
 
         self.gradient_checkpointing = False
 
-    def forward(self, hidden_states, res_hidden_states_tuple, temb=None, upsample_size=None, scale: float = 1.0):
+    def forward(
+        self,
+        hidden_states,
+        res_hidden_states_tuple,
+        temb=None,
+        upsample_size=None,
+        scale: float = 1.0,
+    ):
+        is_freeu_enabled = getattr(self.config, "s1", None) and getattr(self.config, "s2", None) and getattr(self.config, "b1", None), getattr(self.config, "b2", None)  
+
         for resnet in self.resnets:
             # pop res hidden states
             res_hidden_states = res_hidden_states_tuple[-1]
             res_hidden_states_tuple = res_hidden_states_tuple[:-1]
+            
+            if is_freeu_enabled:
+                # --------------- FreeU code -----------------------
+                # Only operate on the first two stages
+                # `fourier_filter` needs to be implemented.
+                if hidden_states.shape[1] == 1280:
+                    hidden_states[:, :640] = hidden_states[:, :640] * self.config["b1"]
+                    res_hidden_states = fourier_filter(res_hidden_states, threshold=1, scale=self.config["s1"])
+                if hidden_states.shape[1] == 640:
+                    hidden_states[:, :320] = hidden_states[:, :320] * self.config["b2"]
+                    res_hidden_states = fourier_filter(res_hidden_states, threshold=1, scale=self.config["s2"])
```

Comment:
```suggestion
                if self.resolution_idx == 0:  # add this to init
                    hidden_states[:, :hidden_states.shape[-1] // 2] = hidden_states[:, :hidden_states.shape[-1] // 2] * self.config["b1"]
                    res_hidden_states = fourier_filter(res_hidden_states, threshold=1, scale=self.config["s1"])
                elif self.resolution_idx == 1:  # add this to init
                    hidden_states[:, :hidden_states.shape[-1] // 2] = hidden_states[:, :hidden_states.shape[-1] // 2] * self.config["b2"]
                    res_hidden_states = fourier_filter(res_hidden_states, threshold=1, scale=self.config["s2"])
```

Not a huge fan of hard-coding values in here - can we instead pass the resolution index to the block to make it more explicit?

Reply:
`resolution_idx` does not seem naturally connecting to me. What does it mean?

Reply:
Yeah maybe `block_idx` is better